src/sfptpd_clock.c

```diff
@@ -2484,4 +2571,163 @@ bool sfptpd_clock_is_active(const struct sfptpd_clock *clock)
 	return clock && !clock->deleted;
 }
 
+int sfptpd_clock_deduplicate(void)
+{
+	struct sfptpd_clock **snapshot;
```

Comment:
nit - `snapshot` is used as a list of the non-Solarflare clocks. In code later on, it was a bit confusing as "snapshot" could also refer to the time. Something like `clock_list` might be easier to parse.